diff -rub MMF2Exts/DarkEdif/Inc/Android/MMFAndroidMasterHeader.hpp NoiseExtension/sdk/inc/Android/MMFAndroidMasterHeader.hpp
--- MMF2Exts/DarkEdif/Inc/Android/MMFAndroidMasterHeader.hpp	2026-02-22 12:59:48.638129335 +0100
+++ NoiseExtension/sdk/inc/Android/MMFAndroidMasterHeader.hpp	2025-10-01 17:16:42.970663056 +0200
@@ -15,7 +15,7 @@
 #include "../Shared/AllPlatformDefines.hpp"
 #include "../Shared/NonWindowsDefines.hpp"
 
-#include <asm-generic\posix_types.h>
+#include <asm-generic/posix_types.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <iomanip>
@@ -838,9 +838,5 @@
 const int REFLAG_ONESHOT = 2;
 
 
-// Defined in DarkEdif.cpp with ASM instructions to embed the binary.
-extern char darkExtJSON[];
-extern unsigned darkExtJSONSize;
-
 // Undo the warning disabling from earlier
 #pragma clang diagnostic pop
Tylko w NoiseExtension/sdk/inc/: FastNoiseLite.h
Tylko w NoiseExtension/sdk/inc/: Html
Tylko w MMF2Exts/DarkEdif/Inc/: iOS
Tylko w MMF2Exts/DarkEdif/Inc/: Mac
diff -rub MMF2Exts/DarkEdif/Inc/Shared/AllPlatformDefines.hpp NoiseExtension/sdk/inc/Shared/AllPlatformDefines.hpp
--- MMF2Exts/DarkEdif/Inc/Shared/AllPlatformDefines.hpp	2026-02-22 12:59:48.638129335 +0100
+++ NoiseExtension/sdk/inc/Shared/AllPlatformDefines.hpp	2025-08-03 21:19:42.693601843 +0200
@@ -78,7 +78,11 @@
 	// Where formatParamIndex is 1-based index of the format param, and dots is the 1-based index of ...
 	// Note class member functions should include the "this" pointer in the indexing.
 	// You can use 0 for dotsParamIndex for vprintf-like format instead.
+    #ifdef __APPLE__
 	#define PrintFHintAfter(formatParamIndex,dotsParamIndex) __printflike(formatParamIndex, dotsParamIndex)
+    #else
+        #define PrintFHintAfter(formatParamIndex,dotsParamIndex) __attribute__((__format__(__printf__, formatParamIndex, dotsParamIndex)))
+    #endif
 #else
 	#define PrintFHintInside /* no op */
 	#define PrintFHintAfter(formatParamIndex,dotsParamIndex) /* no op */
diff -rub MMF2Exts/DarkEdif/Inc/Shared/Edif.hpp NoiseExtension/sdk/inc/Shared/Edif.hpp
--- MMF2Exts/DarkEdif/Inc/Shared/Edif.hpp	2026-02-22 12:59:48.638129335 +0100
+++ NoiseExtension/sdk/inc/Shared/Edif.hpp	2025-07-31 19:18:20.087565891 +0200
@@ -12,13 +12,15 @@
 	#include "..\Windows\MMFWindowsMasterHeader.hpp"
 	extern HINSTANCE hInstLib;
 #elif defined (__ANDROID__)
-	#include "..\Android\MMFAndroidMasterHeader.hpp"
+	#include "../Android/MMFAndroidMasterHeader.hpp"
 #elif defined (__APPLE__)
-#if MacBuild == 0
+    #if MacBuild == 0
 	#include "../iOS/MMFiOSMasterHeader.hpp"
-#else
+    #else
 	#include "../Mac/MMFMacMasterHeader.hpp"
-#endif
+    #endif
+#elif defined(__wasi__)
+    #include "../Html/MMFHtmlMasterHeader.hpp"
 #endif
 
 class Extension;
@@ -185,9 +187,14 @@
 		static void DetachJVMAccessForThisThread();
 		// Gets JNIEnv * for this thread, or null.
 		static JNIEnv * GetJNIEnvForThisThread();
-#else
+#elif defined(__APPLE__)
 		Runtime(Extension* ext, void * const objCExtPtr);
 		void * curCEvent;
+
+#elif defined(__wasi__)
+		Runtime(Extension* ext);
+#else
+    #error Unsupported platform.
 #endif
 
 		~Runtime();
Tylko w MMF2Exts/DarkEdif/Inc/Windows: ksnd.h
Tylko w MMF2Exts/DarkEdif/Inc/Windows: zconf.h
Tylko w MMF2Exts/DarkEdif/Inc/Windows: zlib.h
